### DISCUSSION TITLE
RunTask without endpoint definition in Elsa Studio

### DISCUSSION THREAD

**Discussion by aleperitesar** [elsa-workflows/elsa-studio#244] [Q&A]
Hi, this is the worfklow.

My problem, the issue, is, where can I declare the endpoint that this certain RunTask has to point? Because I have 2 different workflow, the drink and the food workflow. The RunTask of the food workflow has to point to https://localhost:5002/api/webhooks/run-task-food AND the RunTask of the drink workflow has to point to https://localhost:5002/api/webhooks/run-task-drink.
Task-Food will handle FoodEvent, and drink-food will handle DrinkEvent, so this is the reason cuz i want do have two different endpoint. 
The Server is the same, so at this moment this configuration will fire the request to the same endpoint, but it can handle only food-event so i can't trigger drink-workflow because it will break.

This is the stupid workflow i'm using as a test:

[workflow-1.json](https://github.com/user-attachments/files/16094613/workflow-1.json)

**Comment by glime-ai** [comment]
To configure different endpoints for the `RunTask` activities in your food and drink workflows, you will need to modify the `appsettings.json` file of your workflow server application to include multiple webhook endpoints, each with a different URL and set of event types.

Here's an example of how you might configure the `appsettings.json` file to include two separate endpoints for food and drink events:

```json
{
  "Webhooks": {
    "Endpoints": [
      {
        "EventTypes": ["FoodEvent"],
        "Url": "https://localhost:5002/api/webhooks/run-task-food"
      },
      {
        "EventTypes": ["DrinkEvent"],
        "Url": "https://localhost:5002/api/webhooks/run-task-drink"
      }
    ]
  }
}
```

In this configuration, the workflow server will send webhook requests to the specified URLs based on the event type that is published. Make sure that the `EventTypes` in your `RunTask` activities within your workflows match the `EventTypes` specified in the `appsettings.json` file.

For the `RunTask` activity in your food workflow, you would publish a `FoodEvent`, and for the drink workflow, you would publish a `DrinkEvent`. This way, the workflow server knows which endpoint to trigger based on the event type.

Please ensure that your external application that handles the tasks is set up to listen to these specific endpoints (`/api/webhooks/run-task-food` and `/api/webhooks/run-task-drink`) and can process the `FoodEvent` and `DrinkEvent` accordingly.

If you need further assistance on how to set up the `RunTask` activity or how to publish specific event types, please refer to the Elsa Workflows documentation or provide additional details about your current workflow configuration.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by aleperitesar** [comment]
@sfmskywalker I can't figure out how to do it even with this Glime answer, I need a little tutorial on how to make it works in server and in studio. Thanks!

**Comment by glime-ai** [comment]
To configure different endpoints for the `RunTask` activities in your food and drink workflows, you will need to modify the `appsettings.json` file of your workflow server application to include multiple webhook endpoints, each with a different URL and associated event types.

Here's an example of how you can set up your `appsettings.json` to include two different endpoints for the food and drink workflows:

```json
{
  "Webhooks": {
    "Endpoints": [
      {
        "EventTypes": ["FoodEvent"],
        "Url": "https://localhost:5002/api/webhooks/run-task-food"
      },
      {
        "EventTypes": ["DrinkEvent"],
        "Url": "https://localhost:5002/api/webhooks/run-task-drink"
      }
    ]
  }
}
```

In this configuration, you define two endpoints within the `Webhooks.Endpoints` array. Each endpoint has an `EventTypes` property that specifies which event type it should handle. The `Url` property specifies the unique URL for handling the specific event type.

In your workflows, you will need to ensure that the `RunTask` activities publish the correct event type (`FoodEvent` for the food workflow and `DrinkEvent` for the drink workflow). The `Elsa.Webhooks` package will then handle the `RunTaskRequest` domain event by firing HTTP requests to the registered webhook matching the event type.

Please note that the event types (`FoodEvent` and `DrinkEvent`) are just examples. You will need to use the actual event types that your workflows are configured to publish.

Remember to update your workflow definitions to publish the correct event types for the `RunTask` activities. If you need further guidance on how to configure the `RunTask` activities to publish specific event types, please refer to the Elsa documentation or provide additional details about your workflow configuration.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
